nncf/quantization/quantize_model.py

```diff
@@ -578,91 +586,57 @@ def compress_weights(
-                f"INT8 modes do not support {', '.join(unsupported_for_int8)} option(s). Set them to None."
-            )
-
-    if ratio is None:
-        ratio = 1
-    if group_size is None:
-        group_size = 128
-    if all_layers is None:
-        all_layers = False
-    if awq is None:
-        awq = False
-    if scale_estimation is None:
-        scale_estimation = False
-    if gptq is None:
-        gptq = False
-    if lora_correction is None:
-        lora_correction = False
-    if ignored_scope is None:
-        ignored_scope = IgnoredScope()
-    if sensitivity_metric is None:
-        sensitivity_metric = (
-            SensitivityMetric.WEIGHT_QUANTIZATION_ERROR
-            if dataset is None
-            else SensitivityMetric.MAX_ACTIVATION_VARIANCE
-        )
-    if backup_mode is None:
-        backup_mode = BackupMode.INT8_ASYM
-    if ratio != 1 and dataset is None and sensitivity_metric != SensitivityMetric.WEIGHT_QUANTIZATION_ERROR:
-        raise nncf.ValidationError(
-            f"Mixed precision selection based on the given sensitivity metric={sensitivity_metric.value} requires "
-            "a dataset, but it's not provided."
-        )
-    if ratio < 0 or ratio > 1:
-        raise nncf.ValidationError(f"The ratio should be between 0 and 1, but ratio={ratio} is specified.")
-    if subset_size is None or subset_size <= 0:
-        raise nncf.ValidationError(f"The subset_size value should be positive, but subset_size={subset_size} is given.")
-
-    if compression_weights_impl is None:
-        raise nncf.UnsupportedBackendError(f"Unsupported type of backend: {backend}")
-
-    return compression_weights_impl(
-        model,
-        dataset,
+    check_user_compression_configuration(
         mode,
+        subset_size,
+        dataset,
         ratio,
         group_size,
-        ignored_scope,
         all_layers,
+        awq,
+        scale_estimation,
+        gptq,
+        lora_correction,
+        ignored_scope,
         sensitivity_metric,
+        backup_mode,
+        advanced_parameters,
+    )
+    weight_compression_configuration = get_weight_compression_configuration(
+        mode,
+        dataset,
+        ratio,
+        group_size,
+        all_layers,
         awq,
-        subset_size,
         scale_estimation,
         gptq,
         lora_correction,
+        ignored_scope,
+        sensitivity_metric,
         backup_mode,
         advanced_parameters,
     )
 
+    if compression_weights_impl is None:
+        raise nncf.UnsupportedBackendError(f"Unsupported type of backend for weight compression: {backend}")
+
+    is_to_cache_statistics = (
+        weight_compression_configuration["advanced_parameters"].statistics_dir_path
+        and not Path(weight_compression_configuration["advanced_parameters"].statistics_dir_path).exists()
+    )
+    if is_to_cache_statistics:
+        from nncf.openvino.quantization.cache_statistics import cache_weight_compression_statistics
```

Comment:
It seems that the stats caching approach is only supported by the OpenVINO backend. If you are not going to implement cross-backend, please move `cache_weight_compression_statistics` to openvino `compression_weights_impl`.